### PR TITLE
Adds missing test to CJK issue

### PR DESCRIPTION
**Pull request by HarelM**
Adds missing test to #1002

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Write tests for all new functionality.

It's not ideal as the image is saved in base64 format and is compared exactly, but I didn't want to complicate the test to use PNG and pixelmatch libraries...

### CODE REVIEW

**Comment by HarelM** [comment]
This is failing as the generated image is the following:
image
Need to understand why this is not working in CI :-/

File: test/integration/browser/browser.test.ts [hunk 1]
```diff
@@ -94,6 +96,82 @@ describe('drag and zoom', () => {
...
             expect(zoom).toBe(2);
         }, 20000);
+
+        test(`${impl.name()} - CJK Characters`, async () => {
+            await newTest(impl);
+            await page.evaluate(() => {
+
+                map.setStyle({
+                    version: 8,
+                    glyphs: 'https://mierune.github.io/fonts/{fontstack}/{range}.pbf',
```
Comment (wipfli) [inline test/integration/browser/browser.test.ts:106]: Interesting stuff @HarelM, thanks for looking into these tests. Is this font dependency something we want to keep or should we move it into the test assets?

File: test/integration/browser/browser.test.ts [hunk 2]
```diff
@@ -2,6 +2,8 @@ import {Browser, BrowserContext, BrowserType, chromium, Page} from 'playwright';
 import address from 'address';
```
Comment (wipfli) [inline test/integration/browser/browser.test.ts:2]: Cool to see these browser tests. Is `playwright` what we decided to use? Does this do the same as `puppeteer` and `cypress`?

Comment (HarelM) [reply test/integration/browser/browser.test.ts:2]: Playwright is the successor of puppeteer. Cypress is more wildly used but needs bigger code changes so for now we are with playwright

File: test/integration/browser/browser.test.ts (hunk 1 above)
Comment (HarelM) [reply test/integration/browser/browser.test.ts:106]: I think this is OK for now as I took it from the reported issue. We can always move it in later if needed.

File: .github/workflows/test-browser.yml [hunk 3]
```diff
@@ -7,8 +7,8 @@ on:
 
 jobs:
   test_browser:
-    name: ubuntu-latest
-    runs-on: ubuntu-latest
+    name: macos-latest
```
Comment (wipfli) [inline .github/workflows/test-browser.yml:10]: Please comment on the OS switch.

**Review by wipfli — APPROVED** [review]
Looks good to me. Thanks!

File: .github/workflows/test-browser.yml (hunk 3 above)
Comment (HarelM) [reply .github/workflows/test-browser.yml:10]: I switch due to the above issue - see the image I added - there's not CJK font in ubuntu CI server I think and this is causing a "missing" characters.
Having said that, the image I get on my mac and the image I get in the CI are not exactly the same as it seems that the image on the mac CI is a bit more bold.
Having said that, the test is checking what it needs and passes on the CI server.

**Comment by wipfli** [comment]
Haha, running the browser tests locally now looks like this: 

image

**Comment by HarelM** [comment]
I know :-) but at least you can use the received test, save it to an html file and see how it looks in the browser.
These were my intentions, although not very user friendly, but a lot less complicated than render tests...

**Comment by birkskyum** [comment]
How come the CI passes when this test fails?

**Comment by wipfli** [comment]
I think so, yes. If I remember correctly, @HarelM chose to use base64 image encoding such that one can easily copy-paste the CI actual result to the expected result in version control...

**Comment by HarelM** [comment]
There are probably better ways to do it, but I wanted to have a CJK test in the CI as it's not possible to do it in the render tests.
Yes, it fails locally unfortunately.
A better way would've been to use pixelMatch like in the render tests I guess, but I didn't want to over complicate the browser test... Feel free to refactor this...

**Pull request merged**